roles/content_library/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+---
+- name: Check Mandatory Variables Are Defined
+  ansible.builtin.assert:
+    that:
+      - content_library_hostname is defined
+      - content_library_username is defined
+      - content_library_password is defined
+      - content_library_name is defined
+    quiet: true
+    fail_msg: Variable must be set when using this role.
+
+- name: Check Datastore Name Is Defined
+  ansible.builtin.fail:
+    msg: content_library_datastore_name is required when library state is 'present'
+  when: content_library_datastore_name is not defined and content_library_state == 'present'
+
+- name: Manage Content Library
```

Comment:
I just wonder why do we need to create such a role? The advantage is that it validate the varibales. But it don't bring any other advantage in compare to `community.vmware.vmware_content_library_manager` module. Why should user choose to use this role if, using module give him better usability in terms of checking return values of the module? I could understand we have such module for vm_management, where we can use both SOAP and REST, and maybe add some other functionality on top of it. But I am not sure how we could extend this role, and we couldn't the module.

Reply:
for a pre-step job template ... otherwise, each one of the AAP users should implement such playbook

Reply:
I had the same thought as you ondra, but after talking with Sean I think theres some group of people who would prefer to only interact with AAP and not necessarily write yaml/know what module they need to use

Reply:
So still, this can be a playbook, which calls the module, not necessary role, with all the READMEs and stuff around it. But role could be extensible in future so it's fine, I just wanted to bring this up, so we don't need to maintain necessarily two same things.

Reply:
Thank you for bringing it up you're right. 
I think that in content you should use roles with README 
but it's a good point (we can extend it in the future)